Flamedriver.asm

```diff
@@ -504,9 +504,9 @@ calcVolume macro
 		or	a								; Is it positive?
 		jp	p, .skip_track_vol				; Branch if yes
 		add	a, (ix+zTrack.Volume)			; Add track's volume to it
-		; TODO: Maybe turn this into a saturation add to prevent clipping?
+		jr	nc, .skip_track_vol
```

Comment:
I am looking over, and this is unfortunately not enough to prevent clipping on all cases. It is also not correct -- going from 7Fh to 81h will have carry clear and will clip anyway. See cfChangeVolume for an example of the "right" way to do this.

Reply:
Thinking about it, the issue you mention in the description is probably caused by zFadeInToPrevious not doing overflow check when adding 40h to volume, along with other things.

Reply:
7Fh to 81h isn't possible: the `jp	p, .skip_track_vol` will ensure only `a` values >= 80h will occur.

Reply:
Hm, I keep forgetting about that check. You are right, of course.